common.js

```diff
@@ -39,7 +39,7 @@ function initMetadataFromRequest(request = {}, port = {}) {
         auth: request.auth.credentials,
         method: request.payload && request.payload.method,
         opcode: request.payload && request.payload.method ? request.payload.method.split('.').pop() : '',
-        mtid: (request.payload && request.payload.id == null) ? 'notification' : 'request',
+        mtid: (request.payload && request.payload.id == null && request.payload.method === null) ? 'notification' : 'request',
```

Comment:
This is not correct. Notifications also have `method`.

Reply:
Usually the handler is not impacted by the mtid. It is up to the requesting system do decide if it wants to wait for the response to a request. I am not sure what you are trying to achieve and why/if mtid matters for your handler.

Reply:
Sorry, this is a bug, which was introduced recently during upgrade to hapi 17. Push needs to happen for both cases. I will fix it now.